examples/kuka_iiwa_arm/iiwa_controller_demo.cc

```diff
@@ -0,0 +1,193 @@
+// Kp and 'Rotational' Kp
+#define K_P 25
+#define K_OMEGA 15
+
+// Kd and 'Rotational' Kd
+#define K_D 1
+#define K_R 0.3
+
+#define NUM_JOINTS 7
+
+#include "drake/common/trajectories/piecewise_polynomial.h"
+#include "drake/examples/kuka_iiwa_arm/iiwa_lcm.h"
+#include "drake/systems/lcm/lcm_publisher_system.h"
+#include "drake/systems/lcm/lcm_subscriber_system.h"
+#include "drake/systems/lcm/lcm_interface_system.h"
+#include "drake/systems/analysis/simulator.h"
+#include "drake/systems/framework/diagram.h"
+#include "drake/systems/framework/diagram_builder.h"
+#include "drake/systems/primitives/trajectory_source.h"
+#include "drake/systems/primitives/constant_vector_source.h"
+#include "drake/lcm/drake_lcm.h"
+#include "drake/lcmt_iiwa_command.hpp"
+#include "drake/lcmt_iiwa_status.hpp"
+#include "drake/multibody/plant/multibody_plant.h"
+#include "drake/multibody/parsing/parser.h"
+
+#include "systems/controllers/endeffector_velocity_controller.h"
+#include "systems/controllers/endeffector_position_controller.h"
+#include "systems/vector_scope.h"
+
+namespace dairlib {
+
+int do_main(int argc, char* argv[]) {
+
+
+  // Creating end effector trajectory
+  // TODO make this modular
+  const std::vector<double> times {0.0, 25.0, 35.0, 45.0, 55.0, 65.0, 75.0, 85.0, 95.0, 105.0, 115};
+
+  std::vector<Eigen::MatrixXd> points(times.size());
+
+  Eigen::Vector3d AS, A0, A1, A2, A3, A4, A5, A6, A7, A8, A9;
+
+  AS << -0.23, -0.2, 0.25;
+  A0 << -0.23, -0.2, 0.25;
+
+  A1 << -0.23, -0.6, 0.25;
+  A2 << 0.23, -0.6, 0.25;
+
+  A3 << 0.23, -0.2, 0.25;
+
+  A4 << 0.23, -0.2, 0.25;
+
+  A5 << 0.23, -0.6, 0.25;
+
+  A6 << -0.23, -0.6, 0.25;
+
+  A7 << -0.23, -0.2, 0.25;
+
+  A8 << -0.23, -0.2, 0.25;
+  A9 << -0.23, -0.2, 0.25;
+
+  points[0] = AS;
+  points[1] = A0;
+  points[2] = A1;
+  points[3] = A2;
+  points[4] = A3;
+  points[5] = A4;
+  points[6] = A5;
+  points[7] = A6;
+  points[8] = A7;
+  points[9] = A8;
+  points[10] = A9;
```

Comment:
There should be a more terse way of initializing this. For example initializing a points matrix and then for looping over each row to initialize the points vector. There's probably a better way to do that. Even here it seems like you only have four unique points so the rest are unnecessary

Reply:
I agree this code is really ugly, but I'm leaving it for now because I'll be making a second pull request with Evan's code soon that automates this and pulls in points from a .csv file without needing to recompile the code every time we want to change the trajectory